diff --git a/src/cmake/externalpackages.cmake b/src/cmake/externalpackages.cmake
index 9835ed398..b484383c5 100644
--- a/src/cmake/externalpackages.cmake
+++ b/src/cmake/externalpackages.cmake
@@ -195,7 +195,6 @@ checked_find_package (OpenCV 3.0
                    DEFINITIONS  -DUSE_OPENCV=1)
 
 # Intel TBB
-set (TBB_USE_DEBUG_BUILD OFF)
 checked_find_package (TBB 2017
                       SETVARIABLES OIIO_TBB
                       PREFER_CONFIG)
