diff --git a/cmake/modules/FindOpenEXR.cmake b/cmake/modules/FindOpenEXR.cmake
index 639a01cf1..d57af1d79 100644
--- a/cmake/modules/FindOpenEXR.cmake
+++ b/cmake/modules/FindOpenEXR.cmake
@@ -56,6 +56,10 @@ if(OPENEXR_INCLUDE_DIR)
   endif()
 endif()
 
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
+  SET(LIB_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+endif()
+
 foreach(OPENEXR_LIB
     Half
     Iex
@@ -70,8 +74,8 @@ foreach(OPENEXR_LIB
     # using both versioned and unversioned names.
     find_library(OPENEXR_${OPENEXR_LIB}_LIBRARY
         NAMES
-            ${OPENEXR_LIB}-${OPENEXR_MAJOR_VERSION}_${OPENEXR_MINOR_VERSION}
-            ${OPENEXR_LIB}
+            ${OPENEXR_LIB}-${OPENEXR_MAJOR_VERSION}_${OPENEXR_MINOR_VERSION}${LIB_POSTFIX}
+            ${OPENEXR_LIB}${LIB_POSTFIX}
         HINTS
             "${OPENEXR_LOCATION}"
             "$ENV{OPENEXR_LOCATION}"
diff --git a/cmake/modules/FindOpenImageIO.cmake b/cmake/modules/FindOpenImageIO.cmake
index 9df47ff23..7d2811b84 100644
--- a/cmake/modules/FindOpenImageIO.cmake
+++ b/cmake/modules/FindOpenImageIO.cmake
@@ -22,6 +22,10 @@
 # language governing permissions and limitations under the Apache License.
 #
 
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
+  SET(LIB_POSTFIX ${CMAKE_DEBUG_POSTFIX})
+endif()
+
 if(UNIX)
     find_path(OIIO_BASE_DIR
             include/OpenImageIO/oiioversion.h
@@ -31,7 +35,7 @@ if(UNIX)
             "/opt/oiio"
     )
     find_path(OIIO_LIBRARY_DIR
-            libOpenImageIO.so
+            libOpenImageIO${LIB_POSTFIX}.so
         HINTS
             "${OIIO_LOCATION}"
             "$ENV{OIIO_LOCATION}"
@@ -49,7 +53,7 @@ elseif(WIN32)
             "$ENV{OIIO_LOCATION}"
     )
     find_path(OIIO_LIBRARY_DIR
-            OpenImageIO.lib
+            OpenImageIO${LIB_POSTFIX}.lib
         HINTS
             "${OIIO_LOCATION}"
             "$ENV{OIIO_LOCATION}"
@@ -81,7 +85,7 @@ foreach(OIIO_LIB
     )
 
     find_library(OIIO_${OIIO_LIB}_LIBRARY
-            ${OIIO_LIB}
+            ${OIIO_LIB}${LIB_POSTFIX}
         HINTS
             "${OIIO_LOCATION}"
             "$ENV{OIIO_LOCATION}"
diff --git a/cmake/modules/FindOpenSubdiv.cmake b/cmake/modules/FindOpenSubdiv.cmake
index 686af9b16..7cbdb0f47 100644
--- a/cmake/modules/FindOpenSubdiv.cmake
+++ b/cmake/modules/FindOpenSubdiv.cmake
@@ -27,15 +27,14 @@ IF(NOT OPENSUBDIV_ROOT_DIR AND NOT $ENV{OPENSUBDIV_ROOT_DIR} STREQUAL "")
     SET(OPENSUBDIV_ROOT_DIR $ENV{OPENSUBDIV_ROOT_DIR})
 ENDIF()
 
-SET(_opensubdiv_FIND_COMPONENTS
-    osdCPU
-)
-if(OPENSUBDIV_USE_GPU)
-    list(APPEND
-        _opensubdiv_FIND_COMPONENTS
-        osdGPU)
+if(CMAKE_BUILD_TYPE STREQUAL Debug)
+  SET(LIB_POSTFIX ${CMAKE_DEBUG_POSTFIX})
 endif()
 
+SET(_opensubdiv_FIND_COMPONENTS osdCPU${LIB_POSTFIX})
+if(OPENSUBDIV_USE_GPU)
+    list(APPEND _opensubdiv_FIND_COMPONENTS osdGPU${LIB_POSTFIX})
+endif()
 
 SET(_opensubdiv_SEARCH_DIRS
     ${OPENSUBDIV_ROOT_DIR}
